File: app/src/main/java/com/mynameismidori/currencypickerexample/MainActivity.java
package com.mynameismidori.currencypickerexample;

import android.os.Bundle;
import androidx.preference.PreferenceManager;
import androidx.fragment.app.FragmentActivity;

public class MainActivity extends FragmentActivity {
    private CurrencyFragment currencyFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        PreferenceManager.setDefaultValues(this, R.xml.preferences, false);
        currencyFragment = new CurrencyFragment();
        currencyFragment.setArguments(getIntent().getExtras());

        getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, currencyFragment).commit();
    }

}

File: app/src/main/java/uk/co/iloveruby/currencypickerexample/kotlin/MainActivity.kt
package uk.co.iloveruby.currencypickerexample.kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager
import com.mynameismidori.currencypickerexample.R

class MainActivity : AppCompatActivity() {
    private lateinit var currencyFragment: CurrencyFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        PreferenceManager.setDefaultValues(this, R.xml.preferences, false)
        currencyFragment = CurrencyFragment()
        currencyFragment.arguments = intent.extras
        supportFragmentManager.beginTransaction()
            .replace(R.id.container, currencyFragment)
            .commit()
    }
}